AutoCollection/WebSnippet.ts

```diff
@@ -68,13 +69,98 @@ class WebSnippet {
     }
 
     private _getWebSnippetIkey(connectionString: string) {
-        const csCode = ConnectionStringParser.parse(connectionString);
-        const iKeyCode = csCode.instrumentationkey || "";
-        if (!ConnectionStringParser.isIkeyValid(iKeyCode)) {
-            this._isIkeyValid = false;
-            Logging.info("Invalid web snippet connection string, web snippet will not be injected.");
+        let iKey = null;
+        try {
+            const csCode = ConnectionStringParser.parse(connectionString);
+            const iKeyCode = csCode.instrumentationkey || "";
+            if (!ConnectionStringParser.isIkeyValid(iKeyCode)) {
+                this._isIkeyValid = false;
+                Logging.info("Invalid web Instrumentation connection string, web Instrumentation is not enabled.");
+            } else {
+                this._isIkeyValid = true;
+                iKey = iKeyCode;
+            }
+        } catch (err) {
+            Logging.info("get web snippet ikey error: " + err);
+        }
+        return iKey;
+    }
+
+    private _getWebInstrumentationReplacedStr() {
+        let configStr = this._getClientWebInstrumentationConfigStr(this._clientWebInstrumentationConfig);
+        let osStr = this._getOsString();
+        let rpStr = this._getResourceProvider();
+        let snippetReplacedStr = `${this._webInstrumentationIkey}\",\r\n${configStr} disableIkeyDeprecationMessage: true,\r\n sdkExtension: \"${rpStr}${osStr}r_n_`;
+        let replacedSnippet = webSnippet.replace("INSTRUMENTATION_KEY", snippetReplacedStr);
+        if (this._clientWebInstrumentationSrc) {
+            return replacedSnippet.replace(`${Constants.WEB_INSTRUMENTATION_DEFAULT_SOURCE}.2.min.js`,this._clientWebInstrumentationSrc);
+        }
+        return replacedSnippet;
+    }
+
+    private _getOsString() {
```

Comment:
We use this line to determine if Linux/Windows in Agent

  const isWindows = process.platform === 'win32';